From 6af327f2aa17e24cf4e2f38754b676157b24a0a5 Mon Sep 17 00:00:00 2001
From: Brian Behlendorf <behlendorf1@llnl.gov>
Date: Tue, 3 Nov 2015 11:04:10 -0800
Subject: [PATCH 02/18] Add isIdle helper

Checks if a buildslave is available and currently idle.  In the case
of a latent buildslave it may be available but not substantiated.  This
helper allows us to determine if a build can be immediately serviced
without the cost of substantiating a new buildslave.

Signed-off-by: Brian Behlendorf <behlendorf1@llnl.gov>
---
 master/buildbot/process/slavebuilder.py | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/master/buildbot/process/slavebuilder.py b/master/buildbot/process/slavebuilder.py
index 86f427d3c..42f050e5b 100644
--- a/master/buildbot/process/slavebuilder.py
+++ b/master/buildbot/process/slavebuilder.py
@@ -72,6 +72,9 @@ class AbstractSlaveBuilder(pb.Referenceable):
         # no slave? not very available.
         return False
 
+    def isIdle(self):
+        return self.isAvailable() and self.state is IDLE
+
     def isBusy(self):
         return self.state not in (IDLE, LATENT)
 
-- 
2.13.6

